### PR TITLE
Parachains para_inherent.rs to FrameV2

### PR DESCRIPTION
**Pull request by ferrell-code**
relates: #2882 

Following the upgrade guidelines here: https://crates.parity.io/frame_support/attr.pallet.html#upgrade-guidelines.

# :warning: Breaking Change :warning: 
From https://crates.parity.io/frame_support/attr.pallet.html#checking-upgrade-guidelines

 >storages now use PalletInfo for module_prefix instead of the one given to decl_storage: Thus any use of this pallet in construct_runtime! should be careful to update name in order not to break storage or to upgrade storage (moreover for instantiable pallet). If pallet is published, make sure to warn about this breaking change.

So users of the `ParaInherent` pallet must be careful about the name they used in `construct_runtime!.` Hence the runtime-migration label, which might not be needed depending on the configuration of the `ParaInherent` pallet.

kusama, rococo and westend use `ParasInherent` in `construct_runtime!`,  so I migrated them to `ParaInherent` 

polkadot does not use `ParaInherent`.

### CODE REVIEW

File: runtime/parachains/src/disputes.rs [hunk 1]
```diff
@@ -1152,6 +1152,9 @@ mod tests {
 		while System::block_number() < to {
 			let b = System::block_number();
 			if b != 0 {
+				// circumvent requirement to have bitfields and headers in block for testing purposes
+				crate::paras_inherent::Included::<Test>::set(Some(()));
+
 				AllPallets::on_finalize(b);
```
Comment (ferrell-code) [inline runtime/parachains/src/disputes.rs:1158]: The addition of ParaInherent in the mock `construct_runtime!` causes the ParaInherent `on_finalize` hook to now fire and will panic as no paras inherent have been entered. I believe this to be reasonable for the disputes tests but always good to double check

**Review by shawntabrizi — APPROVED** [review]

File: runtime/parachains/src/disputes.rs (hunk 1 above)
Comment (shawntabrizi) [reply runtime/parachains/src/disputes.rs:1158]: looks fine to me

File: runtime/parachains/src/paras_inherent.rs [hunk 2]
```diff
@@ -73,30 +69,89 @@ decl_error! {
...
+							disputes: Vec::new(),
+							parent_header: inherent_data.parent_header,
+						}
+					},
+				};
+
+			Some(Call::enter(inherent_data))
+		}
+
+		fn is_inherent(call: &Self::Call) -> bool {
```
Comment (KiChjang) [inline runtime/parachains/src/paras_inherent.rs:141]: This function seems to be unused, I think we can remove it?

**Review by KiChjang — APPROVED** [review]

**Comment by KiChjang** [comment]
bot merge

**Pull request merged**